src/threat_model/rust_client_threat_model.md

```diff
@@ -0,0 +1,134 @@
+prevent the client application from accessing Parsec functionality.
+
+### Client’s cryptographic keys - AS4
+
+Data stored in the configuration file, read at each reloading of Parsec to instantiate components.
+
+**Confidentiality** : (private) key material is sensitive data and should not be exposed.
+
+**Integrity** : neither public nor private key data should be modifiable by attackers.
+
+**Availability** : the client must always be able to use their cryptographic material.
+
+## Attackers
+
+Each dataflow is analysed from an attacker’s perspective using STRIDE method. Nothing is supposed
+on the different components.
+
+In the following tables are present the type of each possible threat, its description, its
+mitigation status and the assets impacted. A threat can be unmitigated (U), mitigated (M) or
+operational (O). The assumptions context applies for all threats but when one of them is
+particularly relevant, it will be noted with ASUM.
+
+### Attacker “Request to Service” - A1
+
+This attacker uses the inter-process endpoint, presumably created by the service, to interfere with
+the communication between client and service.
+
+|   | Description                                                                                                                                       | Mitigation | Assets        |
+|---|---------------------------------------------------------------------------------------------------------------------------------------------------|------------|---------------|
+| S | An attacker impersonates the service by establishing an IPC endpoint that spoofs the one used by Parsec to exfiltrate any data found in requests. | O-0, U-1   | AS1, AS2, AS4 |
+| T | The attacker modifies the data sent or received by the client while it is in transit through the IPC.                                             | M-0, U-1   | AS2, AS4      |
+| R | The service denies having sent responses to or received some requests from the client.                                                            | U-3        |               |
+| I | An attacker obtains requests sent by the client by sniffing the IPC traffic or spoofing the endpoint.                                             | M-0, U-1   | AS2, AS4      |
+| D | An attacker could remove the IPC endpoint or block traffic through it.                                                                            | O-1, U-1   | AS4           |
+
+### Attacker “Library Memory” - A2
+
+This attacker gains access to memory released by the client library, containing potentially
+sensitive information.
+
+|   | Description                                                 | Mitigation | Assets   |
+|---|-------------------------------------------------------------|------------|----------|
+| I | An attacker reads requests and responses from memory.       | U-4        | AS2, AS4 |
+| I | An attacker reads out the application identity from memory. | U-4        | AS1      |
+
+## Unmitigations
```

Comment:
Only 1, 3 and 4 are used, maybe the others can be deleted and a new ordering done?

Reply:
Same for the Mitigations and OM, not all are used.

Reply:
I think I should try and find ways to add attackers that are covered by these mitigations, since the (un)mitigations seem like they should be mentioned in relation to something